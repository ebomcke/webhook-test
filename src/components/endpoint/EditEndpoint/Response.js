import React, { Suspense } from 'react';
import { Form, Message, Icon, Loader, Dimmer, Segment } from 'semantic-ui-react';
const JsonEditor = React.lazy(() =>
  import(/* webpackChunkName: "JsonEditor" */ '../../base/JsonEditor'),
);

const Response = ({ endpointEditor, nextButton, renderMessage }) => (
  <div>
    {!!renderMessage && (
      <Message icon>
        <Icon name="help" />
        <Message.Content>
          <Message.Header>This is optional</Message.Header>
          <p>
            If you're not sure what the default response should be you can
            safely leave this default value.
          </p>
        </Message.Content>
      </Message>
    )}
    <Form>
      <Form.Field>
        <Form.Select
          fluid
          name="defaultResponseStatusCode"
          onChange={(event, data) =>
            endpointEditor.onStatusCodeChange(data.value)
          }
          label="Status code"
          options={[
            { key: 1, text: '200 - OK', value: 200 },
            { key: 2, text: '201 - Created', value: 201 },
          ]}
          value={endpointEditor.endpoint.defaultResponseStatusCode}
        />
      </Form.Field>
      <Form.Field>
        <label>Response body</label>
        <Suspense
          fallback={
            <Segment>
              <Dimmer active inverted>
                <Loader size="big" inverted>
                  Loading
                </Loader>
              </Dimmer>
            </Segment>
          }
        >
          <JsonEditor
            height="300px"
            value={endpointEditor.endpoint.defaultResponseBody}
            onChange={endpointEditor.onResponseBodyChange}
          />
        </Suspense>
      </Form.Field>
      {nextButton()}
    </Form>
  </div>
);

export default Response;
