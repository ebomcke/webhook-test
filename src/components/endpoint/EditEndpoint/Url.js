import React from 'react';
import { compose } from 'recompose';
import { withAccount } from '../../../contexts/Account';
import { Form, Input } from 'semantic-ui-react';

const Url = ({ endpointEditor, account, nextButton, disabled }) => (
  <Form>
    <Form.Field required>
      <label>URL</label>
      <Input
        disabled={disabled}
        label={`${process.env.API_URL}/webhook/${account.organisationName}/`}
        placeholder="my/custom/url"
        name="path"
        value={endpointEditor.endpoint.path}
        onChange={(event, data) => endpointEditor.onUrlChange(data.value)}
      />
    </Form.Field>
    {!!nextButton && nextButton()}
  </Form>
);

export default compose(withAccount)(Url);
