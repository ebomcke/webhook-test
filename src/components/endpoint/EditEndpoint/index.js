import React from 'react';
import { Form, Header } from 'semantic-ui-react';
import Url from './Url';
import Response from './Response';

const EditEndpoint = ({ endpointEditor }) => (
  <div>
    <Header as="h1">Edit endpoint</Header>
    <Url endpointEditor={endpointEditor} disabled={true} />
    <p />
    <Response
      renderMessage={false}
      endpointEditor={endpointEditor}
      nextButton={() => (
        <Form.Button name="endpoint-editor-save" onClick={endpointEditor.save}>
          Save
        </Form.Button>
      )}
    />
  </div>
);

export const EditEndpointUrl = Url;
export const EditEndpointResponse = Response;

export default EditEndpoint;
