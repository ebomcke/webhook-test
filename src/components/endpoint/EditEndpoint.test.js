import React from 'react';
import { create } from 'react-test-renderer';
import EditEndpoint from './EditEndpoint';

const endpointEditor = {
  save: jest.fn(),
  endpoint: {
    defaultResponseStatusCode: 200,
    defaultResponseBody: JSON.stringify({
      status: 'OK',
    }),
    path: 'unit-test-path',
  },
};

describe('Rendering tests', () => {
  test('Basic', () => {
    const editEndpoint = create(
      <EditEndpoint endpointEditor={endpointEditor} />,
    );
    expect(editEndpoint.toJSON()).toMatchSnapshot();
  });
});
