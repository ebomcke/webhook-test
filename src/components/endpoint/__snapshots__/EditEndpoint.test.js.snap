// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering tests Basic 1`] = `
<div>
  <h1
    className="ui header"
  >
    Edit endpoint
  </h1>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="required field"
    >
      <label>
        URL
      </label>
      <div
        className="ui disabled labeled input"
      >
        <div
          className="ui label label"
          onClick={[Function]}
        >
          https://webhook-tester-1547141127436.appspot.com/webhook/unit-test-org/
        </div>
        <input
          disabled={true}
          name="path"
          onChange={[Function]}
          placeholder="my/custom/url"
          tabIndex={-1}
          type="text"
          value="unit-test-path"
        />
      </div>
    </div>
  </form>
  <p />
  <div>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <div
          className="field"
        >
          <label>
            Status code
          </label>
          <div
            aria-expanded={false}
            className="ui fluid selection dropdown"
            name="defaultResponseStatusCode"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <div
              aria-live="polite"
              className="text"
              role="alert"
            >
              200 - OK
            </div>
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
            <div
              className="menu transition"
            >
              <div
                aria-checked={true}
                aria-selected={true}
                className="active selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  200 - OK
                </span>
              </div>
              <div
                aria-checked={false}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  201 - Created
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="field"
      >
        <label>
          Response body
        </label>
        <div>
          This is a placeholder for the JSON editor.
          <pre>
            <code>
              {"height":"300px","value":"{\\"status\\":\\"OK\\"}"}
            </code>
          </pre>
        </div>
      </div>
      <div
        className="field"
      >
        <button
          className="ui button"
          name="endpoint-editor-save"
          onClick={[Function]}
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
